compute_tools/src/spec.rs

```diff
@@ -231,9 +230,13 @@ pub fn handle_role_deletions(node: &ComputeNode, client: &mut Client) -> Result<
 fn reassign_owned_objects(node: &ComputeNode, role_name: &PgIdent) -> Result<()> {
     for db in &node.spec.cluster.databases {
         if db.owner != *role_name {
-            let db_name_encoded = format!("/{}", encode(&db.name));
-            let db_connstr = node.connstr.replacen("/postgres", &db_name_encoded, 1);
-            let mut client = Client::connect(&db_connstr, NoTls)?;
+            let mut connstr = node.connstr.clone();
+            connstr
+                .path_segments_mut()
+                .map_err(|_| anyhow::anyhow!("invalid connstr: url is not a base"))?
+                .pop()
+                .push(&db.name);
```

Comment:
Maybe just use `set_path` https://docs.rs/url/latest/url/struct.Url.html#method.set_path ?

You anyway assumes that dbname is at the end, but `set_path` at least will preserve `?` parameters if we ever use them

Reply:
The current code will preserve parameters as well. See the example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8d252b4a9a1508e863e4e6fbc8cab2f5 

But I agree, set_path better describes the intention

Reply:
Yeah, really, `[path_segments_mut](https://docs.rs/url/latest/url/struct.Url.html#method.path_segments_mut)` only manipulates with path segments. For some reason I thought that it will split the whole url into segments, sorry, then doesn't matter much